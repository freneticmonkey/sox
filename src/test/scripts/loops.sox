print "testing loops";

print "Testing While";
var initial = 0;

while initial < 10 {
    initial = initial + 1;
    print initial;
}

print "Testing For";
for var i = 0; i < 10; i = i + 1 {
    print i;
}

print "Testing While Break @ 5"
initial = 0;
while initial < 10 {
    initial = initial + 1;
    print initial;
    if initial == 5 {
        break;
    }
}

print "Testing For Break @ 3";
for var i = 0; i < 10; i = i + 1 {
    print i;
    if i == 3 {
        break;
    }
}