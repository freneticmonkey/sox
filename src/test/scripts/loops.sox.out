testing loops
Testing While
1
2
3
4
5
6
7
8
9
10
Testing For
0
1
2
3
4
5
6
7
8
9
Testing While Break @ 5
1
2
3
4
5
Testing For Break @ 3
0
1
2
3
Testing While Continue @ 5
----------
1
while loop end: 
1
----------
2
while loop end: 
2
----------
3
while loop end: 
3
----------
4
while loop end: 
4
----------
5
----------
6
while loop end: 
6
----------
7
while loop end: 
7
----------
8
while loop end: 
8
----------
9
while loop end: 
9
----------
10
while loop end: 
10
Testing For Continue @ 3
----------
0
loop end: 
0
----------
1
loop end: 
1
----------
2
loop end: 
2
----------
3
----------
4
loop end: 
4
----------
5
loop end: 
5
----------
6
loop end: 
6
----------
7
loop end: 
7
----------
8
loop end: 
8
----------
9
loop end: 
9
